Cargo.toml

```diff
@@ -19,6 +20,10 @@ time = { version = "0.3", features = ["parsing", "formatting"] }
 thiserror = "1.0"
 http = "0.2.5"
 rand = "0.8.4"
+serde = { version = "1.0.140", features = ["derive"], optional = true }
+
+[features]
+serde = ["dep:serde"]
```

Comment:
You can remove this. By making the dependency optional you are implicitly adding a `serde` feature.